internal/cloud/realm/auth.go

```diff
@@ -12,35 +12,80 @@ type loginPayload struct {
 	PrivateAPIKey string `json:"apiKey"`
 }
 
-// AuthResponse is the Realm login response
-type AuthResponse struct {
+// Session is the Realm login response
+type Session struct {
 	AccessToken  string `json:"access_token"`
 	RefreshToken string `json:"refresh_token"`
 }
 
-func (c *client) Authenticate(publicAPIKey, privateAPIKey string) (AuthResponse, error) {
-	path := adminAPI + "/auth/providers/mongodb-cloud/login"
+// UserProfile contains all of the roles associated with a user
+type UserProfile struct {
+	Roles []Role `json:"roles"`
+}
+
+// Role contains a GrouID field that maps to a project ID
+type Role struct {
+	GroupID string `json:"group_id"`
+}
+
+// AllGroupIDs returns all available group ids for a given user
+func (pd UserProfile) AllGroupIDs() []string {
+	groupIDs := []string{}
+	set := make(map[string]bool)
+	for _, role := range pd.Roles {
+		if !set[role.GroupID] {
+			if role.GroupID == "" {
+				continue
+			}
+			groupIDs = append(groupIDs, role.GroupID)
+			set[role.GroupID] = true
+		}
+	}
+
+	return groupIDs
+}
+
+var (
+	authenticatePath = adminAPI + "/auth/providers/mongodb-cloud/login"
+)
+
+func (c *client) Authenticate(publicAPIKey, privateAPIKey string) (Session, error) {
+	res, resErr := c.doJSON(http.MethodPost, authenticatePath, loginPayload{publicAPIKey, privateAPIKey}, api.RequestOptions{})
+	if resErr != nil {
+		return Session{}, resErr
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return Session{}, UnmarshalServerError(res)
+	}
 
-	opts, optsErr := api.JSONRequestOptions(loginPayload{publicAPIKey, privateAPIKey})
-	if optsErr != nil {
-		return AuthResponse{}, optsErr
+	dec := json.NewDecoder(res.Body)
+	defer res.Body.Close()
+
+	var session Session
+	if err := dec.Decode(&session); err != nil {
+		return Session{}, err
 	}
+	return session, nil
+}
 
-	res, resErr := c.do(http.MethodPost, path, opts)
+func (c *client) GetUserProfile() (UserProfile, error) {
```

Comment:
i'd like to consolidate the naming b/w `UserProfile` and `AuthProfile` (between path variables and response structures, it'd be nice to have a unified language here)

Reply:
Done, change to AuthProfile since this logic lives in Auth